library/pool.c

```diff
@@ -1184,46 +1197,129 @@ int dnet_io_init(struct dnet_node *n, struct dnet_config *cfg)
 	return err;
 }
 
+int dnet_server_backend_init(struct dnet_node *n, size_t backend_id)
```

Comment:
This function looks exactly the same as code in `dnet_server_io_init()`
Also why wouldn't use tree here instead of array?